iroha_p2p/src/peer.rs

```diff
@@ -490,15 +577,20 @@ where
 #[async_trait::async_trait]
 impl<T, K, E> ContextHandler<StopSelf> for Peer<T, K, E>
 where
-    T: Encode + Decode + Send + Sync + Clone + 'static,
+    T: Debug + Encode + Decode + Send + Sync + Clone + 'static,
     K: KeyExchangeScheme + Send + 'static,
     E: Encryptor + Send + 'static,
 {
     type Result = ();
 
     async fn handle(&mut self, ctx: &mut Context<Self>, message: StopSelf) {
-        if message.0 == self.connection_id {
-            debug!("Stopping self");
+        // Zero means stopping whole network
+        if message.0 == 0 || message.0 == self.connection_id {
```

Comment:
Oof this seems a lot like pure C style code. Use enum in this case:
```
enum StopSelf {
 Id(..),
 AllNetwork
 }
 ```

Reply:
Done.